File: 프로그래머스/lv2/12946. 하노이의 탑/하노이의 탑.js
function solution_(n, from, through, to) {
    if(n == 1)
        return [[from,to]]
    let result = []
    result = result.concat(solution_(n - 1, from, to, through))
    result.push([from, to])
    result = result.concat(solution_(n - 1, through, from, to))
    return result
}

function solution(n) {
    return solution_(n,1,2,3)
}

File: 프로그래머스/lv2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    def solution_(n, from_, through, to):
        if n == 1:
            return [[from_, to]]
        result = []
        result.extend(solution_(n - 1, from_, to, through))
        result.append([from_, to])
        result.extend(solution_(n - 1, through, from_, to))
        return result

    return solution_(n, 1, 2, 3)
